pkg/kn/commands/source/human_readable_flags.go

```diff
@@ -85,3 +101,148 @@ func printSourceTypesList(sourceTypesList *unstructured.UnstructuredList, option
 	}
 	return rows, nil
 }
+
+// printSource populates a single row of source list table
+func printSource(source unstructured.Unstructured, options printers.PrintOptions) ([]metav1beta1.TableRow, error) {
+	name := source.GetName()
+	sourceType := source.GetKind()
+	sourceTypeName := getSourceTypeName(source)
+	sink := findSink(source)
+	ready := isReady(source)
+
+	row := metav1beta1.TableRow{
+		Object: runtime.RawExtension{Object: &source},
+	}
+
+	if options.AllNamespaces {
+		row.Cells = append(row.Cells, source.GetNamespace())
+	}
+
+	row.Cells = append(row.Cells, name, sourceType, sourceTypeName, sink, ready)
+	return []metav1beta1.TableRow{row}, nil
+}
+
+// printSourceList populates the source list table rows
+func printSourceList(sourceList *unstructured.UnstructuredList, options printers.PrintOptions) ([]metav1beta1.TableRow, error) {
+	rows := make([]metav1beta1.TableRow, 0, len(sourceList.Items))
+
+	sort.SliceStable(sourceList.Items, func(i, j int) bool {
+		return sourceList.Items[i].GetName() < sourceList.Items[j].GetName()
+	})
+	for _, item := range sourceList.Items {
+		row, err := printSource(item, options)
+		if err != nil {
+			return nil, err
+		}
+
+		rows = append(rows, row...)
+	}
+	return rows, nil
+}
+
+func findSink(source unstructured.Unstructured) string {
```

Comment:
I think the `sink` should be always at the place in a source (as its a DuckType "Sinkable" probably ;-). So you could just navigate to those field into the unstructured source itself. But as mentioned elsewhere it might be better to move that kind of extraction allready when the list is queried from the server and introduce a view object that is just printed here. Putting too much of logic in the print functions makes it harder to maintain later.